buildlib/tools/coverity.sh

```diff
@@ -63,7 +63,7 @@ run_coverity() {
 	if [ "${ucx_build_type}" == "devel" ]; then
 		cov-manage-emit --dir $cov_build --tu-pattern "file('.*/test/gtest/common/googletest/*')" delete
 	fi
-	cov-analyze --jobs $parallel_jobs $COV_OPT --security --concurrency --dir $cov_build
+	cov-analyze --jobs $parallel_jobs $COV_OPT --disable SLEEP --security --concurrency --dir $cov_build
```

Comment:
can we add a local coverity suppression as a code comment instead?

Reply:
i had no luck with local suppression; this checker only understands annotations at the lock acquisition location, and not at the sleep location. btw - maybe sched_yield() instead of sleep will be enough?

Reply:
what is the coverity error?
sched_yield - not sure is needed - we call ibv_reg_mr sys call anyway and AFAIK kernel should call schedule before returning from syscall

Reply:
might sleep while holding lock
[output.log](https://github.com/user-attachments/files/15987033/output.log)

Reply:
seems it is 
lock: "pthread_rwlock_wrlock" locks "rcache->pgt_lock
but we're not supposed to hold rcache lock during memory registration
so maybe need some fix in rcache?

Reply:
following out-of-band talk, will try to remove the sleep and relay on syscall resched